orangebeard/OrangebeardClient.py

```diff
@@ -0,0 +1,431 @@
+        self.__call_events[temp_uuid] = start_step_event
+
+        parent_event_uuid = start_step.testUUID if start_step.parentStepUUID is None \
+            else start_step.parentStepUUID
+
+        parent_event = self.__call_events[parent_event_uuid]
+
+        self.__event_loop.run_until_complete(self.__exec_start_step(start_step, temp_uuid, parent_event))
+        return temp_uuid
+
+    def finish_step(self, step_uuid: UUID, finish_step: FinishStep) -> None:
+        """
+        Finish a step.
+
+        Args:
+            step_uuid (UUID): The UUID of the step to be finished.
+            finish_step (FinishStep): The FinishStep object containing information about finishing the step.
+        """
+        finish_step_event = asyncio.Event()
+        temp_uuid = uuid.uuid4()
+        self.__call_events[temp_uuid] = finish_step_event
+
+        parent_event = self.__call_events[step_uuid]
+        asyncio.ensure_future(self.__exec_finish_step(step_uuid, finish_step, temp_uuid, parent_event))
+
+    def log(self, log: Log) -> UUID:
+        """
+        Send a log.
+
+        Args:
+            log (Log): The Log object containing information to store.
+
+        Returns:
+            UUID: The UUID associated with the log.
+        """
+        log_event = asyncio.Event()
+        temp_uuid = uuid.uuid4()
+        self.__call_events[temp_uuid] = log_event
+
+        parent_event_uuid = log.testUUID if log.stepUUID is None \
+            else log.stepUUID
+        parent_event = self.__call_events[parent_event_uuid]
+
+        asyncio.ensure_future(self.__exec_log(log, temp_uuid, parent_event))
+        return temp_uuid
+
+    def send_attachment(self, attachment: Attachment) -> UUID:
+        """
+        Send an attachment. (to attach to a log)
+
+        Args:
+            attachment (Attachment): The Attachment object containing information about the attachment.
+
+        Returns:
+            UUID: The UUID associated with the attachment that was sent.
+        """
+        attachment_event = asyncio.Event()
+        temp_uuid = uuid.uuid4()
+        self.__call_events[temp_uuid] = attachment_event
+
+        parent_event = self.__call_events[attachment.AttachmentMetaData.logUUID]
+        asyncio.ensure_future(self.__exec_send_attachment(attachment, temp_uuid, parent_event))
+        return temp_uuid
+
+    async def __make_api_request(self, method: str, uri: str, data: Serializable = None):
+        if self.__connection_with_orangebeard_is_valid:
+            async with self.__client.request(method, uri,
+                                             data=data.to_json() if data is not None else None) as response:
+                if 200 <= response.status < 300:
+                    try:
+                        return await response.json()
+                    except ContentTypeError:
+                        return None
+                else:
+                    print(f'Error Sending: {data.to_json()} to {uri}')
```

Comment:
Wat gebeurt hier als data == None?

Reply:
Dan print 'ie "Error sending: null to {uri}". Heb ik getest omdat ik iets van een NPE verwachtte, maar kennelijk is json.dumps heel vergeeflijk